pkg/controller/datadogagentdeployment/datadogagentdeployment_controller.go

```diff
@@ -263,3 +279,87 @@ func (r *ReconcileDatadogAgentDeployment) updateStatusIfNeeded(logger logr.Logge
 
 	return result, currentError
 }
+
+// sendDatadogMetrics establishes a connection with the Datadog API
+// it starts sending deployment metrics once the connection is validated
+func (r *ReconcileDatadogAgentDeployment) sendDatadogMetrics(ctx context.Context) {
+	retryTicker := time.NewTicker(r.datadogMetricsForwarder.RetryInterval)
+RetryLoop:
```

Comment:
I'm sure we can do it without it.

have you look into the package: `k8s.io/apimachinery/pkg/util/wait` it provide some usefull function that implement for you the ticker, stop. force instance the `Poll` function seams to do the work
https://github.com/kubernetes/apimachinery/blob/master/pkg/util/wait/wait.go#L313